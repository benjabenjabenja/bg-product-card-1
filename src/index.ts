/**
 * ACA SE EXPORTAN TODOS LOS COMPONENTES, HOOKS, STYLES, Y LO QUE SEA QUE SE QUIERA EXPORTAR
 */

// COMPONENTES
export * from './components'

