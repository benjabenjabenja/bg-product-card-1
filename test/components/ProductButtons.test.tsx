import React from 'react';
import { render } from '@testing-library/react';
import { ProductButtons } from '../../src';


describe('ProductButtons', () => {

    test('should render component buttons with custom className', () => {
        const { container } = render(<ProductButtons className="custom-class" />);

        expect(container).toMatchSnapshot();
        expect(container.querySelector('.custom-class')).toBeDefined();
    })

    test('should render component buttons with custom style', () => {
        const { container } = render(<ProductButtons style={{ backgroundColor: 'red' }} />);
        
        expect(container).toMatchSnapshot();
        expect(container.querySelector('.buttonsContainer')?.getAttribute('style')).toContain('background-color: red;');
    })

})

