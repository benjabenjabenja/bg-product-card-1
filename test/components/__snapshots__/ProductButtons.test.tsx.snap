// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductButtons should render component buttons with custom className 1`] = `
<div>
  <div
    class="buttonsContainer custom-class"
  >
    <button
      class="buttonMinus"
    >
       - 
    </button>
    <div
      class="countLabel"
    >
       
       
    </div>
    <button
      class="buttonAdd false "
    >
       + 
    </button>
  </div>
</div>
`;

exports[`ProductButtons should render component buttons with custom style 1`] = `
<div>
  <div
    class="buttonsContainer undefined"
    style="background-color: red;"
  >
    <button
      class="buttonMinus"
    >
       - 
    </button>
    <div
      class="countLabel"
    >
       
       
    </div>
    <button
      class="buttonAdd false "
    >
       + 
    </button>
  </div>
</div>
`;
